### PR TITLE
Internal review fixes

### PR DESCRIPTION
**Pull request by woocash2**
## Description

Apply fixes suggested by the internal review / audit. Some notes:
* Haven't applied suggestions with regard to using `amounts[amounts.len() - 1]`. A panic instead of an error is not a big issue, and if fixing this, I would feel pressure to also change every use of `some_array[0]` to `some_array.first()`, which would be too much.
* Haven't written new tests - could be done in a separate PR, while this version already gets deployed to the dev env.
* Haven't acted upon the last finding - still thinking what to do (I would actually prefer to do nothing, but maybe leaving a comment wouldn't hurt, but definitely don't want to refactor the code so that amounts are returned by the swap methods).

## Testing

All drink tests pass.

### CODE REVIEW

File: amm/contracts/router_v2/stable_pool.rs [hunk 1]
```diff
@@ -129,7 +124,7 @@ impl StablePool {
                 self.transfer_tokens_back(&amounts, to, wnative_idx)?;
                 res
             }
-            None => {
+            Err(_) => {
```
Comment (krzysztofziobro) [inline amm/contracts/router_v2/stable_pool.rs:127]: To be 100% clean it should be done only when matching a specific variant and propagate error upwards if some other error is present (same for the second method).

From what I see it's not possible to get any other error than `InvalidToken`, so your call

**Review by krzysztofziobro — APPROVED** [review]

File: amm/contracts/router_v2/stable_pool.rs (hunk 1 above)
Comment (JanKuczma) [reply amm/contracts/router_v2/stable_pool.rs:127]: NIT: Maybe even cleaner would be to change `fn wnativre_idx(&self, wnative: AccountId)` to return `Option<usize>`? After this refactor there's only one place where we want to return `InvalidToken` error when wnative is not present.

**Review by JanKuczma — APPROVED** [review]

File: amm/contracts/router_v2/stable_pool.rs (hunk 1 above)
Comment (woocash2) [reply amm/contracts/router_v2/stable_pool.rs:127]: I like @JanKuczma's suggestion - if `wnative_idx` returns an error, it always has to be `InvalidToken`, and inside the fn we can see that it converts the `Option` to `Result` as the last operation. So let's just return an `Option` and wherever `wnative_idx` was used to potentially throw an error, just map the `Option` to `Result` there.

Comment (krzysztofziobro) [reply amm/contracts/router_v2/stable_pool.rs]: Yes, that's even better

**Pull request merged**